test/extensions/retry/host/omit_host_metadata/config_test.cc

```diff
@@ -0,0 +1,84 @@
+#include "envoy/extensions/retry/host/omit_host_metadata/v3/omit_host_metadata_config.pb.h"
+#include "envoy/registry/registry.h"
+#include "envoy/upstream/retry.h"
+
+#include "extensions/retry/host/omit_host_metadata/omit_host_metadata.h"
+#include "extensions/retry/host/well_known_names.h"
+
+#include "test/mocks/upstream/mocks.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using namespace testing;
+
+namespace Envoy {
+namespace Extensions {
+namespace Retry {
+namespace Host {
+namespace {
+
+TEST(OmitHostsRetryPredicateTest, PredicateTest) {
+  auto factory = Registry::FactoryRegistry<Upstream::RetryHostPredicateFactory>::getFactory(
+      RetryHostPredicateValues::get().OmitHostMetadataPredicate);
+
+  ASSERT_NE(nullptr, factory);
+
+  envoy::extensions::retry::host::omit_host_metadata::v3::OmitHostMetadataConfig config;
+  auto* metadata_match = config.mutable_metadata_match();
```

Comment:
Can you add a test case for when this is empty as well? Since we're now handling that case

Reply:
Done. Added a small check in shouldSelectAnotherHost to check for empty config since metadataLabelMatch returns true in case config is empty.